react/features/prejoin/components/country-picker/CountryPicker.js

```diff
@@ -0,0 +1,245 @@
+// @flow
+
+import React, { Component } from 'react';
+import InlineDialog from '@atlaskit/inline-dialog';
+import { setDialOutCountry, setDialOutNumber } from '../../actions';
+import CountrySelector from './CountrySelector';
+import CountryDropDown from './CountryDropdown';
+import { getDialOutCountry, getDialOutNumber } from '../../functions';
+import { connect } from '../../../base/redux';
+import { getCountryFromDialCodeText } from '../../utils';
+
+const PREFIX_REG = /^(00)|\+/;
+
+type Props = {
+
+    /**
+     * The country to dial out to.
+     */
+    dialOutCountry: { name: string, dialCode: string, code: string },
+
+    /**
+     * The number to dial out to.
+     */
+    dialOutNumber: string,
+
+    /**
+     * Handler used when user presses 'Enter'.
+     */
+    onSubmit: Function,
+
+    /**
+     * Sets the dial out number.
+     */
+    setDialOutNumber: Function,
+
+    /**
+     * Sets the dial out country.
+     */
+    setDialOutCountry: Function,
+};
+
+type State = {
+
+    /**
+     * If the country picker is open or not.
+     */
+    isOpen: boolean,
+
+    /**
+     * The value of the input.
+     */
+    value: string
+}
+
+/**
+ * This component displays a country picker with an input for the phone number.
+ */
+class CountryPicker extends Component<Props, State> {
+    /**
+     * A React ref to the HTML element containing the {@code input} instance.
+     */
+    inputRef: Object;
+
+    /**
+     * Initializes a new {@code CountryPicker} instance.
+     *
+     * @inheritdoc
+     */
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isOpen: false,
+            value: ''
+        };
+        this.inputRef = React.createRef();
+        this._onChange = this._onChange.bind(this);
+        this._onDropdownClose = this._onDropdownClose.bind(this);
+        this._onCountrySelectorClick = this._onCountrySelectorClick.bind(this);
+        this._onEntryClick = this._onEntryClick.bind(this);
+        this._onKeyPress = this._onKeyPress.bind(this);
+    }
+
+    _onChange: (Object) => void;
```

Comment:
We usually put these private methods (starting with _) after the public ones, like `render` and `componentDidMount` (same for other components)

Reply:
Done.